File: Formação Lógica de Programação/JavaScript/Escrevendo as Classes de Um Jogo.js
class Heroi {
  constructor(nome, idade, tipo) {
    this.nome = nome; // Variável de escopo da classe
    this.idade = idade;
    this.tipo = tipo;
  }

  atacar() {
    // Variável local para armazenar o tipo de ataque
    let ataque = "";

    switch (this.tipo.toLowerCase()) {
      case "mago":
        ataque = "magia";
        break;
      case "guerreiro":
        ataque = "espada";
        break;
      case "monge":
        ataque = "artes marciais";
        break;
      case "ninja":
        ataque = "shuriken";
        break;
      default:
        ataque = "um ataque misterioso"; // Caso um tipo não mapeado seja inserido
    }

    console.log(`O ${this.tipo} atacou usando ${ataque}`);
  }
}

// Criando instâncias (Objetos) da classe Heroi
const heroi1 = new Heroi("Arthur", 30, "guerreiro");
const heroi2 = new Heroi("Merlin", 150, "mago");
const heroi3 = new Heroi("Shao Lin", 45, "monge");
const heroi4 = new Heroi("Hattori", 25, "ninja");

const listaDeHerois = [heroi1, heroi2, heroi3, heroi4];

console.log("--- Batalha Iniciada ---\n");

for (let i = 0; i < listaDeHerois.length; i++) {
  listaDeHerois[i].atacar();
}

File: Formação Lógica de Programação/Python/Escrevendo as Classes de Um Jogo.py
class Heroi:
    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        # Em Python, if/elif/else faz o papel do 'switch' tradicional
        tipo_formatado = self.tipo.lower()
        
        if tipo_formatado == "mago":
            ataque = "magia"
        elif tipo_formatado == "guerreiro":
            ataque = "espada"
        elif tipo_formatado == "monge":
            ataque = "artes marciais"
        elif tipo_formatado == "ninja":
            ataque = "shuriken"
        else:
            ataque = "um ataque misterioso"

        print(f"O {self.tipo} atacou usando {ataque}")

heroi1 = Heroi("Arthur", 30, "guerreiro")
heroi2 = Heroi("Merlin", 150, "mago")
heroi3 = Heroi("Shao Lin", 45, "monge")
heroi4 = Heroi("Hattori", 25, "ninja")

lista_de_herois = [heroi1, heroi2, heroi3, heroi4]

print("--- Batalha Iniciada ---\n")

for heroi in lista_de_herois:
    heroi.atacar()